### DISCUSSION TITLE
Using Docker to complete the course

### DISCUSSION THREAD

**Discussion by evelynnmimijae** [Cyfrin/security-and-auditing-full-course-s23#33] [Q&A] [answered]
I am borrowing a laptop and want to avoid installing a bunch of programs. Is it possible to use Docker as a container to complete this course? If so, what are the key differences I should be aware of?

**Comment by PatrickAlphaC** [comment]
Hmm... There would be a lot of weird docker stuff to do there. 

Maybe you could try github codespaces? 

or use gitpod for each section? 

Ie: 

https://gitpod.io/#github.com/Cyfrin/security-and-auditing-full-course-s23

**Reply by evelynnmimijae** [reply]
Hmmm, thanks for the recommendations. 

(Please excuse the late reply. Holiday craze at work!)

**Comment by cromewar** [comment]
The main tool used on security course is Foundry and you can securely install in on docker using this guide: https://book.getfoundry.sh/tutorials/foundry-docker

I don't see why you wouldn't be able to create a container with all the dependencies you need for smart contract development and auditing with docker.

**Reply by evelynnmimijae** [reply]
Thanks for the insight. I decided against Docker because as @PatrickAlphaC pointed out, it's too complicated. Also, I realized that I need to take the Foundry course first. Maybe that's why Docker is too complicated for me.

**Reply by cromewar** [reply, marked as answer]
@evelynnmimijae yeah, I think is still possible to configure docker properly to work for this but it requires a lot of tinkering and mess things up is really easy. Glad you decided to take the Foundry course, have a wonderful year!!

**Reply by evelynnmimijae** [reply]
Likewise 🙂
